src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemAlignmentUtils.java

```diff
@@ -57,9 +67,15 @@ public static SAMRecord applyAlignment( final String readName, final byte[] base
                         }
                     }
                 }
-                bases = Arrays.copyOfRange(bases, alignment.getSeqStart(), alignment.getSeqEnd());
+            }
+            if (tmpCigar.getFirstCigarElement().getOperator() == CigarOperator.H ||
+                    tmpCigar.getLastCigarElement().getOperator() == CigarOperator.H) {
+                final int leftHardClip = tmpCigar.getFirstCigarElement().getOperator() == CigarOperator.H ? tmpCigar.getFirstCigarElement().getLength() : 0;
```

Comment:
I think I haven't explained well enough what I'm proposing.  I propose that you replace lines 73 through 78 with this code:
```
            bases = Arrays.copyOfRange(bases, alignment.getSeqStart(), alignment.getSeqEnd());
            if ( quals.length != 0 ) {
                quals = Arrays.copyOfRange(quals, alignment.getSeqStart(), alignment.getSeqEnd());
            }
```